front/src/modules/timesStops/hooks/useOutputTableData.ts

```diff
@@ -1,157 +1,126 @@
+            ...matchingPathStep,
+            opId: op.id,
+            name: op.extensions?.identifier?.name,
+            ch: op.extensions?.sncf?.ch,
           };
-        }),
-    [selectedTrainSchedule, suggestedOperationalPoints]
-  );
-  const pathStepsByUic = keyBy(
-    pathStepsWithOpPointIndices,
-    (pathStep) => `${pathStep.uic}-${pathStep.secondary_code}`
-  );
+        }
+
+        // compute arrival time
+        const matchingReportTrainIndex = simulatedTrain.positions.findIndex(
+          (position) => position === op.position
+        );
+
+        const time =
+          matchingReportTrainIndex === -1
+            ? interpolateValue(simulatedTrain, op.position, 'times')
+            : simulatedTrain.times[matchingReportTrainIndex];
+        const calculatedArrival = new Date(startDatetime.getTime() + time);
 
-  const outputTableData = useMemo(() => {
-    const pathStepRows = pathStepsWithPositionOnPath.map((pathStep) => {
-      const schedule = scheduleByAt[pathStep.id];
-      const scheduleData = computeScheduleData(schedule);
-      return { ...pathStep, ...formatScheduleData(scheduleData) };
-    });
-
-    const suggestedOpRows = suggestedOperationalPoints.map((sugOpPoint, sugOpIndex) => {
-      const opPoint = operationalPoints.find((op) => op.id === sugOpPoint.opId);
-      if (!opPoint) {
-        return sugOpPoint;
-      }
-      const nextOpPoint = findNextScheduledOpPoint(
-        operationalPoints,
-        pathStepsWithOpPointIndices,
-        sugOpIndex
-      );
-      const pathStepKey = `${sugOpPoint.uic}-${sugOpPoint.ch}`;
-
-      if (pathStepKey in pathStepsByUic) {
-        const pathStepId = pathStepsByUic[pathStepKey].id || '';
-        const schedule = scheduleByAt[pathStepId];
-        const scheduleData = computeScheduleData(schedule);
-        const formattedScheduleData = formatScheduleData(scheduleData);
-        const marginsData = nextOpPoint
-          ? computeMargins(simulatedTrain, opPoint, nextOpPoint, selectedTrainSchedule, pathStepId)
-          : null;
-        const calculatedArrival = checkAndFormatCalculatedArrival(scheduleData, opPoint.time);
         return {
-          ...sugOpPoint,
-          ...formattedScheduleData,
-          receptionSignal: schedule?.reception_signal,
-          calculatedArrival,
-          calculatedDeparture:
-            opPoint.duration > 0
-              ? secToHoursString(opPoint.time + opPoint.duration, { withSeconds: true })
-              : '',
-          ...marginsData,
+          isWaypoint: false,
+          opId: op.id,
+          name: op.extensions?.identifier?.name,
+          ch: op.extensions?.sncf?.ch,
+          calculatedArrival: dateToHHMM(calculatedArrival, { withSeconds: true }),
         };
-      }
-
-      return {
-        ...sugOpPoint,
-        calculatedArrival: secToHoursString(opPoint.time, { withSeconds: true }),
-      };
-    });
+      }),
+    [pathProperties.operationalPoints]
```

Comment:
I think the usememo is lying a bit about its dependencies
```suggestion
    - [pathProperties.operationalPoints]
    + [pathProperties.operationalPoints, pathStepRows, simulatedTrain, selectedTrainSchedule]
```

Reply:
You're right, thanks !!